vyper/semantics/types/function.py

```diff
@@ -714,6 +728,10 @@ def _parse_decorators(
         if decorator.get("id") == "nonreentrant":
             if nonreentrant_node is not None:
                 raise StructureException("nonreentrant decorator is already set", nonreentrant_node)
+            if is_interface:
```

Comment:
i think we can move this validation up to the caller

Reply:
lifted all non-general checks to callers in: https://github.com/vyperlang/vyper/pull/4178/commits/b8389641c6020b2b78728e4f78e6b663450601e1